ci-operator/step-registry/gather-network/OWNERS

```diff
@@ -0,0 +1,37 @@
+approvers:
+- smarterclayton
+- wking
+- stevekuznetsov
```

Comment:
Let's make this list only contain the people who should actually review and approve this specific step!

Reply:
@stevekuznetsov Who are the people for the approve list?

Reply:
@pecameron SDN team people. Maybe just copy the OWNERS file that Jacob added for the hybrid overlay bits in the step registry.

Reply:
Dropped non-sdnteam members from approvers